pkg/store/proxy_heap.go

```diff
@@ -111,24 +112,30 @@ func (d *dedupResponseHeap) Next() bool {
 func chainSeriesAndRemIdenticalChunks(series []*storepb.SeriesResponse) *storepb.SeriesResponse {
 	chunkDedupMap := map[uint64]*storepb.AggrChunk{}
 
+	buf := make([]byte, 8)
+	hasher := xxhash.New()
```

Comment:
I think we have a hasher pool somewhere in the repo, maybe we should use that instead?